src/vsphere_cpi/lib/cloud/vsphere/resources/vm.rb

```diff
@@ -331,6 +335,17 @@ def move_disks_to_old_path(disks_to_move)
         end
       end
 
+      def move_disks_to_current_path(disks_to_move, disk_path)
```

Comment:
This one is a bit tricky, and why I split the methods in the first place.   Because if you do a blanket
```
if disk_path.nil?
  original_disk_path = get_old_disk_filepath(disk.key)
  dest_filename = original_disk_path.match(/^\[[^\]]+\] (.*)/)[1]
else
  # Use current disk path
```
This changes the logic to *always* move disks to the current disk path if it's passed in, which will happen on any CPI `detach_disk` call (the only time it's not passed in during `delete_vm` where we detach all data disks remaining, usually non-BOSH persistent disks, e.g. Kubernetes PVs).   This means the movement logic will ignore the vApp property altogether and never move disks back to their old datastore (the current behavior).

With the combined `move_disks` method, we can preserve the existing logic if we pass the `force_disk_path` (whether set or nil) in per disk to ensure the appropriate behavior.  I introduced a Struct in `vm.rb` to accomplish this.   Have a look.